File: app/src/main/java/net/mbonnin/arcanetracker/PicassoBarRequestHandler.java
package net.mbonnin.arcanetracker;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;

import com.esotericsoftware.kryo.io.Input;
import com.squareup.picasso.Picasso;
import com.squareup.picasso.Request;
import com.squareup.picasso.RequestHandler;

import java.io.IOException;
import java.io.InputStream;

public class PicassoBarRequestHandler extends RequestHandler {
    @Override
    public boolean canHandleRequest(Request data) {
        if (data.uri.getScheme().equals("bar")) {
            return true;
        }

        return false;
    }

    @Override
    public Result load(Request request, int networkPolicy) throws IOException {

        String cardId = request.uri.getHost();
        Picasso.LoadedFrom loadedFrom = Picasso.LoadedFrom.DISK;

        InputStream inputStream = ArcaneTrackerApplication.getContext().getAssets().open("bars/" + cardId + ".webp");
        if (inputStream == null) {
            throw  new IOException();
        }

        Bitmap b = BitmapFactory.decodeStream(inputStream);
        inputStream.close();

        return new Result(b, loadedFrom);
    }
}

File: app/src/main/java/net/mbonnin/arcanetracker/PicassoBarRequestHandler.kt
package net.mbonnin.arcanetracker

import android.graphics.BitmapFactory
import com.squareup.picasso.Picasso
import com.squareup.picasso.Request
import com.squareup.picasso.RequestHandler
import java.io.IOException

class PicassoBarRequestHandler : RequestHandler() {
    override fun canHandleRequest(data: Request): Boolean {
        return data.uri.scheme == "bar"

    }

    @Throws(IOException::class)
    override fun load(request: Request, networkPolicy: Int): RequestHandler.Result {

        val cardId = request.uri.host
        val loadedFrom = Picasso.LoadedFrom.DISK

        val inputStream = HDTApplication.context.getAssets().open("bars/$cardId.webp") ?: throw IOException()

        val b = BitmapFactory.decodeStream(inputStream)
        inputStream.close()

        return RequestHandler.Result(b, loadedFrom)
    }
}
